app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt

```diff
@@ -56,95 +71,123 @@ fun HomeScreen(
     HomeScreenContent(
         showMyCoinsLoading = showMyCoinsLoading,
         showTrendingCoinsLoading = showTrendingCoinsLoading,
+        isRefreshing = rememberRefreshing,
         myCoins = myCoins,
         trendingCoins = trendingCoins,
         onMyCoinsItemClick = viewModel.input::onMyCoinsItemClick,
         onTrendingItemClick = viewModel.input::onTrendingCoinsItemClick
-    )
+    ) { viewModel.refresh() }
 }
 
+@OptIn(ExperimentalMaterialApi::class)
 @Suppress("LongParameterList")
 @Composable
 private fun HomeScreenContent(
     showMyCoinsLoading: IsLoading,
     showTrendingCoinsLoading: IsLoading,
+    isRefreshing: IsLoading,
     myCoins: List<CoinItemUiModel>,
     trendingCoins: List<CoinItemUiModel>,
     onMyCoinsItemClick: (CoinItemUiModel) -> Unit,
-    onTrendingItemClick: (CoinItemUiModel) -> Unit
+    onTrendingItemClick: (CoinItemUiModel) -> Unit,
+    onRefreshing: () -> Unit
 ) {
-    Surface {
-        Column(
-            modifier = Modifier
-                .fillMaxSize()
-                .systemBarsPadding()
-        ) {
-            LazyColumn {
-                item {
-                    Text(
-                        modifier = Modifier
-                            .fillMaxWidth()
-                            .padding(top = Dp16),
-                        text = stringResource(id = R.string.home_title),
-                        textAlign = TextAlign.Center,
-                        style = Style.semiBold24(),
-                        color = MaterialTheme.colors.textColor
-                    )
-                }
-
-                item {
-                    PortfolioCard(
-                        modifier = Modifier.padding(start = Dp16, top = Dp40, end = Dp16)
-                    )
-                }
-
-                item {
-                    MyCoins(
-                        showMyCoinsLoading = showMyCoinsLoading,
-                        coins = myCoins,
-                        onMyCoinsItemClick = onMyCoinsItemClick
-                    )
-                }
+    val refreshingState = rememberPullRefreshState(
+        refreshing = isRefreshing,
+        onRefresh = onRefreshing,
+        refreshThreshold = PullRefreshDefaults.RefreshThreshold,
+        refreshingOffset = PullRefreshDefaults.RefreshThreshold
+    )
 
-                item {
-                    Box(
-                        modifier = Modifier
-                            .fillMaxWidth()
-                            .padding(start = Dp16, top = Dp24, end = Dp16, bottom = Dp16)
-                    ) {
+    Surface {
+        Box(modifier = Modifier.pullRefresh(refreshingState)) {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .systemBarsPadding()
+                    .pullRefresh(refreshingState)
```

Comment:
@hoangnguyen92dn May I know why we need to add `.pullRefresh(refreshingState)` here as we already add it in the `box` scope 🙏

Reply:
@Wadeewee Good catch 👍 It's redundant, we could remote it.